code_samples/back_office/suggestions/src/EventSubscriber/MySuggestionEventSubscriber.php

```diff
@@ -0,0 +1,75 @@
+<?php declare(strict_types=1);
+
+namespace App\EventSubscriber;
+
+use Ibexa\Contracts\ProductCatalog\ProductServiceInterface;
+use Ibexa\Contracts\ProductCatalog\Values\Product\ProductQuery;
+use Ibexa\Contracts\ProductCatalog\Values\Product\Query\Criterion;
+use Ibexa\Contracts\Search\Event\BuildSuggestionCollectionEvent;
+use Ibexa\Contracts\Search\Model\Suggestion\ContentSuggestion;
+use Psr\Log\LoggerAwareTrait;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class MySuggestionEventSubscriber implements EventSubscriberInterface
+{
+    use LoggerAwareTrait;
+
+    private ProductServiceInterface $productService;
+
+    public function __construct(ProductServiceInterface $productService)
+    {
+        $this->productService = $productService;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            BuildSuggestionCollectionEvent::class => ['onBuildSuggestionCollectionEvent', -1],
+        ];
+    }
+
+    public function onBuildSuggestionCollectionEvent(BuildSuggestionCollectionEvent $event): BuildSuggestionCollectionEvent
+    {
+        $suggestionQuery = $event->getQuery();
+        $suggestionCollection = $event->getSuggestionCollection();
+
+        $text = $suggestionQuery->getQuery();
+        $words = explode(' ', preg_replace('/\s+/', ' ', $text));
+        $limit = $suggestionQuery->getLimit();
+
+        try {
+            $productQuery = new ProductQuery(null, new Criterion\LogicalOr([
+                new Criterion\ProductName("$text*"),
+                new Criterion\ProductCode($words),
+                new Criterion\ProductType($words),
+            ]), [], 0, $limit);
+            $searchResult = $this->productService->findProducts($productQuery);
+
+            if ($searchResult->getTotalCount()) {
+                $maxScore = 0.0;
+                /** @var \Ibexa\Contracts\Search\Model\Suggestion\ContentSuggestion $suggestion */
+                foreach ($suggestionCollection as $suggestion) {
+                    $maxScore = max($suggestion->getScore(), $maxScore);
+                }
+
+                /** @var \Ibexa\ProductCatalog\Local\Repository\Values\Product $result */
+                foreach ($searchResult as $result) {
+                    $content = $result->getContent();
+
+                    $contentSuggestion = new ContentSuggestion(
```

Comment:
For Products, you should create a new class ProductSuggestion that extends ```\Ibexa\Contracts\Search\Model\Suggestion\Suggestion```

Reply:
I have added a ProductSuggestion, its normalizer and its renderer in 49e10acea0de0de97192a3d77eb2551ede3884e8...27f514ce24b0c2367bfaafe99a986ec50338b25e. I still ,need to move the HTML template from JS to HTML/Twig. @kisztof I may need help for that.

Reply:
I finally found how to use templates in fbc6c0e9